core/certificate.go

```diff
@@ -16,18 +16,29 @@
 
 package core
 
+import (
+	"crypto"
+)
+
+type NetworkNode interface {
+	GetNodeRef() *RecordRef
+	GetPublicKey() crypto.PublicKey
+}
+
 // Certificate interface provides methods to manage keys
+//go:generate minimock -i github.com/insolar/insolar/core.Certificate -o ../testutils -s _mock.go
 type Certificate interface {
+	NetworkNode
+
 	GetRole() NodeRole
 	GetRootDomainReference() *RecordRef
 	SetRootDomainReference(ref *RecordRef)
 	GetBootstrapNodes() []BootstrapNode
 }
 
-// BootstrapNode holds info about bootstrap nodes
-type BootstrapNode struct {
-	PublicKey   string `json:"public_key"`
-	Host        string `json:"host"`
-	NetworkSign []byte `json:"network_sign"`
-	NodeSign    []byte `json:"node_sign"`
+//go:generate minimock -i github.com/insolar/insolar/core.BootstrapNode -o ../testutils -s _mock.go
+type BootstrapNode interface {
```

Comment:
DiscoveryMeta?

Reply:
Maybe `DiscoveryNode` and `GetDiscoveryNodes()`?

Reply:
Yeah, should be fine.